R/fmelt.R

```diff
@@ -31,8 +33,23 @@ melt.data.table <- function(data, id.vars, measure.vars, variable.name = "variab
     measure.vars = patterns(pats, cols=cols)
   }
   if (is.list(measure.vars) && length(measure.vars) > 1L) {
-    if (length(value.name) == 1L)
-      value.name = paste(value.name, seq_along(measure.vars), sep="")
+    meas.nm = names(measure.vars)
+    if (is.null(meas.nm)) {
+      # user-provided or default stub
+      if (length(value.name) == 1L) {
+        value.name = paste0(value.name, seq_along(measure.vars))
+      }
+    } else {
+      if (length(value.name) > 1L) {
+        warning("'value.name' provided in both 'measure.vars'",
+                "and 'value.name argument'; value provided in",
+                "'measure.vars' is given precedence.")
+      }
+      if (any(is.na(meas.nm)) || !all(nzchar(meas.nm))) {
+        stop("Please provide a name to each element of 'measure.vars'.")
```

Comment:
`anyNA(meas.nm)`

Reply:
@HughParsonage good spot; in fact this is the last surviving instance:

`grep 'any(is.na' R/*`